LibraBFT/Impl/Properties/VotesOnce.agda

```diff
@@ -40,21 +40,31 @@ open import LibraBFT.Yasm.Yasm ℓ-RoundManager ℓ-VSFP ConcSysParms
                                Handle.InitHandler.InitAndHandlers
                                PeerCanSignForPK PeerCanSignForPK-stable
 open        Structural impl-sps-avp
+import      LibraBFT.Impl.Handle.InitProperties as IP
 
 -- This module proves the two "VotesOnce" proof obligations for our handler.
 
 module LibraBFT.Impl.Properties.VotesOnce (𝓔 : EpochConfig) where
 
 newVote⇒lv≡
   : ∀ {pre : SystemState}{pid s' acts v m pk}
-    → ReachableSystemState pre
-    → StepPeerState pid (msgPool pre) (initialised pre)
-        (peerStates pre pid) (s' , acts)
-    → v ⊂Msg m → send m ∈ acts → (sig : WithVerSig pk v)
-    → Meta-Honest-PK pk → ¬ (∈BootstrapInfo-impl fakeBootstrapInfo (ver-signature sig))
-    → ¬ MsgWithSig∈ pk (ver-signature sig) (msgPool pre)
-    → LastVoteIs s' v
-newVote⇒lv≡ _ (step-init initSucc uni) _ send∈acts = ⊥-elim (obm-dangerous-magic' "Use the Contract for the init handler.")
+  → ReachableSystemState pre
+  → StepPeerState pid (msgPool pre) (initialised pre) (peerStates pre pid) (s' , acts)
+  → v ⊂Msg m
+  → send m ∈ acts
+  → (sig : WithVerSig pk v)
+  → Meta-Honest-PK pk
+  → ¬ (∈BootstrapInfo-impl fakeBootstrapInfo (ver-signature sig))
+  → ¬ MsgWithSig∈ pk (ver-signature sig) (msgPool pre)
+  → LastVoteIs s' v
+newVote⇒lv≡ {pre}{pid}{s'}{v = v}{m}{pk} preach (step-init handler-pid-bsi≡just-rm×acts uni) v⊂m send∈acts sig hpk ¬bootstrap ¬mws∈pool
+  with IP.initHandlerSpec.contract pid fakeBootstrapInfo handler-pid-bsi≡just-rm×acts
```

Comment:
I understand why you use these long variable names while developing, but I think they do more harm than good for readability, so I'd prefer you use shorter names that capture the important intuition but don't spell out so much detail.  Up to you do decide whether to change them.